src/data/bedrock/block/convert/BlockStateReader.php

```diff
@@ -349,6 +350,17 @@ public function readWallConnectionType(string $name) : ?WallConnectionType{
 		};
 	}
 
+	/** @throws BlockStateDeserializeException */
+	public function readDripleafTiltType() : DripleafTiltType{
+		return match($type = $this->readString(BlockStateNames::BIG_DRIPLEAF_TILT)){
+			StringValues::BIG_DRIPLEAF_TILT_NONE => DripleafTiltType::NONE(),
+			StringValues::BIG_DRIPLEAF_TILT_UNSTABLE => DripleafTiltType::UNSTABLE(),
+			StringValues::BIG_DRIPLEAF_TILT_PARTIAL_TILT => DripleafTiltType::PARTIAL(),
+			StringValues::BIG_DRIPLEAF_TILT_FULL_TILT => DripleafTiltType::FULL(),
+			default => throw $this->badValueException(BlockStateNames::BIG_DRIPLEAF_TILT, $type),
+		};
+	}
```

Comment:
I'm not really clear what we're gaining by putting this here, as opposed to in the `BlockStateToObjectDeserializer` directly. It's only used once.